alica_engine/src/engine/RunningPlan.cpp

```diff
@@ -570,7 +563,13 @@ void RunningPlan::activate()
     if (isBehaviour()) {
         _basicBehaviour->start(this);
     } else if (_activeTriple.abstractPlan) {
-        _ae->editPlanPool().startPlan(*this);
+        if (!_basicPlan || _basicPlan->getId() != _activeTriple.abstractPlan->getId()) {
+            if (_basicPlan) {
+                _basicPlan->stop();
+            }
+            _basicPlan = _ae->getRuntimePlanFactory().create(_activeTriple.abstractPlan->getId(), dynamic_cast<const Plan*>(_activeTriple.abstractPlan));
+        }
+        _basicPlan->start(this);
```

Comment:
is it possible for the same plan to be reactivated? If so we would be calling init again without call onterminate. I think if basic plan exists it should always be stopped & then started (created if necessary)

Reply:
How was that handled earlier? I didn't see any explicit plan/behaviour stops in case of reactivation.

Reply:
There was no handling in running plan & I'm not sure if successive calls to BasicBehaviour::start() will be made by the engine. However, if they are made, the behaviour till now is that BasicBehaviour ignores such calls, i.e. onInit is not called again. But with this refactor  init will be called again which would mean that we violate the condition that init will never be called without terminating the previous run

Reply:
I fixed it, see commit msg for commit: f127db9a7ffa9cfba12abb09dc44f7b1b16e9715